client/my-sites/patterns/hooks/use-pattern-categories.ts

```diff
@@ -0,0 +1,75 @@
+import { useQuery, UseQueryOptions } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+import type { Category } from 'calypso/my-sites/patterns/types';
+
+export const PATTERN_CATEGORIES = [
+	'featured',
+	'intro',
+	'about',
+	// 'buttons',
+	// 'banner',
+	// 'query',
+	'blog',
+	'posts',
+	// 'call-to-action',
+	// 'columns',
+	// 'coming-soon',
+	'contact',
+	'footer',
+	'forms',
+	'gallery',
+	'header',
+	// 'link-in-bio',
+	// 'media',
+	'newsletter',
+	// 'podcast',
+	'portfolio', // For page patterns only in v1
+	// 'quotes',
+	'services',
+	'store',
+	// 'team',
+	'testimonials', // Reused as "Quotes"
+	// 'text',
+];
+
+export function getPatternCategoriesQueryOptions(
+	locale: string,
+	siteId: undefined | number,
+	queryOptions: Omit< UseQueryOptions< Category[] >, 'queryKey' > = {}
+): UseQueryOptions< Category[] > {
+	return {
+		queryKey: [ locale, siteId, 'pattern-library', 'categories' ],
+		queryFn() {
+			return wpcom.req.get(
+				`/sites/${ encodeURIComponent( siteId ?? '' ) }/block-patterns/categories`,
+				{
+					apiNamespace: 'wp/v2',
+					_locale: locale,
```

Comment:
The locale is automatically inserted on the client, but we need to manually specify this query parameter for SSR.